integrations/weaviate/src/haystack_integrations/document_stores/weaviate/document_store.py

```diff
@@ -505,6 +505,38 @@ def delete_documents(self, document_ids: List[str]) -> None:
         weaviate_ids = [generate_uuid5(doc_id) for doc_id in document_ids]
         self.collection.data.delete_many(where=weaviate.classes.query.Filter.by_id().contains_any(weaviate_ids))
 
+    def delete_all_documents(self, *, recreate_index: bool = False, deletion_batch_size: int = 1000) -> None:
+        """
+        Deletes all documents in a collection.
+        If recreate_index is False, it keeps the collection but deletes documents iteratively.
```

Comment:
Add an empty line between the first sentence and the rest of the docs. That's how it's on haystack, we want at some point to also add the same linting rules for docstrings in integrations.